api/helpers/actions.js

```diff
@@ -2,17 +2,24 @@
 var _ = require('lodash');
 var defaultLog = require('winston').loggers.get('default');
 
-exports.publish = async function (o) {
+exports.publish = async function (o, isProject = false) {
     return new Promise(function (resolve, reject) {
+      // Need project specific logic to handle legislation keys
         // Object wasn't already published?
-        if (!o.read.includes('public')) {
-            var newReadArray = o.read;
+        let newReadArray;
+        if (isProject && !o[o.currentLegislationYear].read.includes('public')) {
```

Comment:
I think the currentLegislationYear in the root is a string of project data keys, i.e. "legislation_2018", etc.. isn't it? I think we'll have to  pull out the year from the string here.

Reply:
we are using the currentLegislationYear as the key here to access the read array inside of o.legislation_2018 or whichever legislation is current

Reply:
right, my bad